### ISSUE TITLE
Integrated terminal closes immediately when set to Windows Bash

### ISSUE THREAD

**Issue by brianblakely** [Microsoft/vscode#10054]
- VSCode Version: 1.3.1
- OS Version: Windows 10.0.14393

Steps to Reproduce:

1.  Enable user setting `"terminal.integrated.shell.windows": "C:\\Windows\\System32\\bash.exe"`
2.  Open integrated terminal

**Comment by mattblang** [comment]
Seeing this as well. I tried to switch back to `cmd.exe` and the same thing happens.

**Comment by brianblakely** [comment]
To clarify, in my case, switching back to cmd/git-bash is successful. But whenever I try pointing to Windows Bash, it fails.

**Comment by mzvast** [comment]
Failed to launch through "c:\\Windows\\System32\\bash.exe" or "c:\\Windows\\sysnative\\bash.exe" .
Same thing in Atom editor.

**Comment by mattblang** [comment]
@mzvast Does switching back to `cmd.exe` work for you?

**Comment by mzvast** [comment]
@brianblakely Nope. Either in Atom.
default value `"terminal.integrated.shell.windows": "C:\\WINDOWS\\system32\\cmd.exe"`
T_T

**Comment by mattblang** [comment]
@mzvast Hmm, does that mean something possibly with `Electron`? Aren't they both built on Electron?

**Comment by mzvast** [comment]
@brianblakely 
Yep,they are  electron-based.
Interesting, when I launch `C:\windows\system32\bash.exe` on WebStorm(which is a java-based IDE),it prompt an error that
`java.io.IOException: Couldn't create PTY`
in which,I think, maybe lies the problem.

**Comment by mattblang** [comment]
Interestingly, if I set `ComSpec` to `%windir%\system32\cmd.exe` then it behaves just like if I try to set it to ``bash.exe`. If I set `ComSpec` to `C:\WINDOWS\system32\cmd.exe` then `cmd.exe` works fine.

**Comment by Tyriar** [comment]
I wrote up some documentation on configuring the terminal for Windows recently https://code.visualstudio.com/docs/editor/integrated-terminal#_configuration

Try using `c:\\Windows\\sysnative\\bash.exe`, and make sure it's installed and you can access it in a regular cmd.exe prompt.

**Comment by mattblang** [comment]
@Tyriar Interesting. If I do that, then the `` CTRL+` `` works, but `` CTRL+SHIFT+C `` does not.

**Comment by Tyriar** [comment]
@mattblang that's because `c:\Windows\sysnative\bash.exe` is a console program instead of a full-blown terminal, you need to run it through cmd.exe. You may be able to write a wrapper batch script that runs cmd.exe and bash.exe inside that, then use that as the external terminal.

**Comment by mattblang** [comment]
@Tyriar Should `c:\windows\system32\bash.exe` not work for both of the hotkeys?

**Comment by brianblakely** [comment]
Thank you for the help @Tyriar

**Comment by Tyriar** [comment]
@mattblang don't know enough about it, but my assumption is that bash.exe needs cmd.exe's UI to be able to run. If you can get it working please share and I'll add to the docs :smiley:

**Comment by mzvast** [comment]
After deleting Code repo in %appdata% and reinstalling VSCODE,then it works.
I've upgraded from 10586,without reset my computer .
Maybe some  cache cause the flashing~
Thanks for all of you.Well done~
:)

**Comment by jamiebarrow** [comment]
So, I'm not on the anniversary update, and pointing to `bash.exe` didn't work, but doing the below seems to be working for me (only just tried it, haven't used it extensively):

```
// settings.json
{
    "terminal.integrated.shell.windows": "C:\\Windows\\System32\\cmd.exe",
    "terminal.integrated.shellArgs.windows": [
        "/k",
        "C:\\Program Files\\Git\\bin\\bash.exe"
    ]
}
```

**Comment by Tyriar** [comment]
@jamiebarrow that's git bash, the version of bash shipped with the git installer, this issue is specifically about the Bash on Ubuntu (WSL) executable.

**Comment by jamiebarrow** [comment]
@Tyriar Ah okay, I don't have the anniversary edition installed, though was wondering if perhaps that would be a workaround if people are getting the same issue I had - trying to run git bash on Win 10 in integrated terminal was giving me the same issue, terminating immediately. Thought it was worth sharing :)

**Comment by Tyriar** [comment]
@jamiebarrow perhaps, thanks for sharing :smiley:

**Comment by davoscollective** [comment]
I came here with this issue, bash integrated terminal just immediately closing upon opening the terminal. I read this thread and then tried opening cmd and running bash. I then realised I didn't have developer mode turned on, and hadn't installed bash correctly.  I fixed it by switching developer mode on, and installing bash by calling it from cmd. 

I originally went to windows features, switched on the linux subsystem, did a restart and then "installed" bash by clicking start, typing bash, and clicking on it, based on some older instructions I read. Then modified the vscode settings. Clearly the wrong process to set it up. I'm just adding this comment because other people might have done the same.
